drivers/FileBase.cpp

```diff
@@ -52,8 +53,8 @@ FileBase::~FileBase() {
     _mutex->unlock();
 
     if (getPathType() == FilePathType) {
-        extern void remove_filehandle(FileLike *file);
-        remove_filehandle(static_cast<FileLike*>(this));
+        extern void remove_filehandle(FileHandle *file);
```

Comment:
We also found ourselves adding global functions to retarget.cpp, and wondering where to declare them.

Given the absence of a separate header file, we decided that FileHandle.h was the correct header file to declare them in - it seems logical that system functions acting on FileHandles like fdopen and remove_filehandle can live there.